server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -843,12 +843,10 @@ public void write(XContentBuilder b) throws IOException {
                 return;
             }
 
-            if (isFragment == false) {
```

Comment:
`isFragment` is not needed, then?

Reply:
It's still needed to not always start an object with a field name. For nested we need to be in control since we make a distinction between single and multi-children.